Bump `sqlx-core` and `sqlx-macros-core` from 0.8.3 ?

- As of submitting this, the latest version of sqlx is `0.8.5`, but the internal packages has breaking changes
  - This itself is expected, documented [here](https://github.com/launchbadge/sqlx/blob/91d26bad4d5e2b05fab1c86d0fbe11586d30f29d/sqlx-core/src/lib.rs#L3-L14)
- When tried to update:

```
error[E0050]: method `begin` has 1 parameter but the declaration in trait `sqlx_core::transaction::TransactionManager::begin` has 2
 --> /home/kanarus/projects/ohkami-rs/sqlx-d1/sqlx-d1-core/src/transaction.rs:8:15
  |
8 |         conn: &mut <Self::Database as sqlx_core::database::Database>::Connection,
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 2 parameters, found 1
  |
  = note: `begin` from trait: `fn(&'conn mut <<Self as TransactionManager>::Database as sqlx_core::database::Database>::Connection, Option<Cow<'static, str>>) -> Pin<Box<(dyn futures_core::Future<Output = Result<(), sqlx_core::Error>> + std::marker::Send + 'conn)>>`

error[E0046]: not all trait items implemented, missing: `get_transaction_depth`
 --> /home/kanarus/projects/ohkami-rs/sqlx-d1/sqlx-d1-core/src/transaction.rs:3:1
  |
3 | impl sqlx_core::transaction::TransactionManager for D1TransactionManager {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `get_transaction_depth` in implementation
  |
  = help: implement the missing item: `fn get_transaction_depth(_: &<<Self as TransactionManager>::Database as sqlx_core::database::Database>::Connection) -> usize { todo!() }`

error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> /home/kanarus/projects/ohkami-rs/sqlx-d1/sqlx-d1-core/src/connection.rs:148:13
    |
148 |             sqlx_core::transaction::Transaction::begin(self)
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^------ argument #2 of type `Option<Cow<'static, str>>` is missing
    |
note: associated function defined here
   --> /home/kanarus/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/sqlx-core-0.8.5/src/transaction.rs:99:12
    |
99  |     pub fn begin(
    |            ^^^^^
help: provide the argument
    |
148 |             sqlx_core::transaction::Transaction::begin(self, /* Option<Cow<'static, str>> */)
    |                                                            +++++++++++++++++++++++++++++++++

Some errors have detailed explanations: E0046, E0050, E0061.
For more information about an error, try `rustc --explain E0046`.
error: could not compile `sqlx-d1-core` (lib) due to 3 previous errors
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,15 +1,21 @@
 [workspace]
 resolver = "3"
-members  = ["sqlx-d1", "sqlx-d1-core", "sqlx-d1-macros"]
-exclude  = ["sandbox"]
+members = ["sqlx-d1", "sqlx-d1-core", "sqlx-d1-macros"]
+exclude = ["sandbox"]
 
 [workspace.package]
-version    = "0.3.1"
-edition    = "2024"
-authors    = ["kanarus <kanarus786@gmail.com>"]
-homepage   = "https://crates.io/crates/sqlx-d1"
+version = "0.4.0"
+edition = "2024"
+authors = ["kanarus <kanarus786@gmail.com>"]
+homepage = "https://crates.io/crates/sqlx-d1"
 repository = "https://github.com/ohkami-rs/sqlx-d1"
-readme     = "README.md"
-keywords   = ["cloudflare", "d1", "sqlx", "database", "db"]
-categories = ["database", "asynchronous", "wasm", "rust-patterns", "network-programming"]
-license    = "MIT"
+readme = "README.md"
+keywords = ["cloudflare", "d1", "sqlx", "database", "db"]
+categories = [
+  "database",
+  "asynchronous",
+  "wasm",
+  "rust-patterns",
+  "network-programming",
+]
+license = "MIT"

--- a/sandbox/Cargo.toml
+++ b/sandbox/Cargo.toml
@@ -1,5 +1,5 @@
 [package]
-name    = "sandbox"
+name = "sandbox"
 version = "0.1.0"
 edition = "2021"
 
@@ -13,7 +13,7 @@ opt-level = "s"
 sqlx-d1 = { path = "../sqlx-d1", features = ["macros", "uuid"] }
 console_error_panic_hook = "0.1"
 thiserror = "2.0"
-worker = { version = "0.7" }
+worker = { version = "0.8" }
 serde = { version = "1.0", features = ["derive"] }
 uuid = { version = "1.16", features = ["serde"] }
 

--- a/sqlx-d1-core/Cargo.toml
+++ b/sqlx-d1-core/Cargo.toml
@@ -27,9 +27,9 @@ DEBUG = []
 
 [dependencies]
 # core dependencies
-sqlx-core = { version = "=0.8.3" }
-worker = { version = "0.7", features = ["d1"] }
-worker-sys = { version = "0.7", features = ["d1"] }
+sqlx-core = { version = "=0.8.6" }
+worker = { version = "0.8", features = ["d1"] }
+worker-sys = { version = "0.8", features = ["d1"] }
 futures-core = { version = "0.3", default-features = false }
 futures-util = { version = "0.3", default-features = false }
 serde = { version = "1.0" }
@@ -39,4 +39,4 @@ rust_decimal = { version = "1", optional = true, default-features = false, featu
 ] }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-sqlx-sqlite = { version = "=0.8.3", features = ["bundled"] }
+sqlx-sqlite = { version = "=0.8.6", features = ["bundled"] }

--- a/sqlx-d1-core/src/connection.rs
+++ b/sqlx-d1-core/src/connection.rs
@@ -163,7 +163,7 @@ const _: () = {
         where
             Self: Sized,
         {
-            sqlx_core::transaction::Transaction::begin(self)
+            sqlx_core::transaction::Transaction::begin(self, None)
         }
 
         fn shrink_buffers(&mut self) {

--- a/sqlx-d1-core/src/transaction.rs
+++ b/sqlx-d1-core/src/transaction.rs
@@ -1,11 +1,16 @@
+use std::borrow::Cow;
+
+use crate::D1Connection;
+
 pub struct D1TransactionManager;
 
 impl sqlx_core::transaction::TransactionManager for D1TransactionManager {
     type Database = crate::D1;
 
-    fn begin(
-        #[allow(unused)] conn: &mut <Self::Database as sqlx_core::database::Database>::Connection,
-    ) -> crate::ResultFuture<'_, ()> {
+    fn begin<'conn>(
+        #[allow(unused)] conn: &'conn mut <Self::Database as sqlx_core::database::Database>::Connection,
+        #[allow(unused)] statement: Option<Cow<'static, str>>,
+    ) -> crate::ResultFuture<'conn, ()> {
         #[cfg(not(target_arch = "wasm32"))]
         {
             unreachable!("Native `TransactionManager` impl")
@@ -44,5 +49,11 @@ impl sqlx_core::transaction::TransactionManager for D1TransactionManager {
 
     fn start_rollback(_: &mut <Self::Database as sqlx_core::database::Database>::Connection) {
         /* do nothing */
+    }
+
+    fn get_transaction_depth(
+        #[allow(unused)] conn: &D1Connection,
+    ) -> usize {
+        1  // dummy since this will never be used because d1 does not support transactions
     }
 }

--- a/sqlx-d1-core/src/transaction.rs
+++ b/sqlx-d1-core/src/transaction.rs
@@ -8,7 +8,8 @@ impl sqlx_core::transaction::TransactionManager for D1TransactionManager {
     type Database = crate::D1;
 
     fn begin<'conn>(
-        #[allow(unused)] conn: &'conn mut <Self::Database as sqlx_core::database::Database>::Connection,
+        #[allow(unused)]
+        conn: &'conn mut <Self::Database as sqlx_core::database::Database>::Connection,
         #[allow(unused)] statement: Option<Cow<'static, str>>,
     ) -> crate::ResultFuture<'conn, ()> {
         #[cfg(not(target_arch = "wasm32"))]
@@ -51,9 +52,7 @@ impl sqlx_core::transaction::TransactionManager for D1TransactionManager {
         /* do nothing */
     }
 
-    fn get_transaction_depth(
-        #[allow(unused)] conn: &D1Connection,
-    ) -> usize {
-        1  // dummy since this will never be used because d1 does not support transactions
+    fn get_transaction_depth(#[allow(unused)] conn: &D1Connection) -> usize {
+        1 // dummy since this will never be used because d1 does not support transactions
     }
 }

--- a/sqlx-d1-macros/Cargo.toml
+++ b/sqlx-d1-macros/Cargo.toml
@@ -1,16 +1,16 @@
 [package]
-name          = "sqlx-d1-macros"
+name = "sqlx-d1-macros"
 documentation = "https://docs.rs/sqlx-d1-macros"
-description   = "proc macros for sqlx-d1 - SQLx for Cloudflare D1"
-version       = { workspace = true }
-edition       = { workspace = true }
-authors       = { workspace = true }
-homepage      = { workspace = true }
-repository    = { workspace = true }
-readme        = { workspace = true }
-keywords      = { workspace = true }
-categories    = { workspace = true }
-license       = { workspace = true }
+description = "proc macros for sqlx-d1 - SQLx for Cloudflare D1"
+version = { workspace = true }
+edition = { workspace = true }
+authors = { workspace = true }
+homepage = { workspace = true }
+repository = { workspace = true }
+readme = { workspace = true }
+keywords = { workspace = true }
+categories = { workspace = true }
+license = { workspace = true }
 
 [package.metadata.docs.rs]
 features = ["derive", "query"]
@@ -19,20 +19,22 @@ features = ["derive", "query"]
 proc-macro = true
 
 [features]
-DEBUG  = []
+DEBUG = []
 derive = ["dep:heck"]
-query  = ["dep:sha2", "dep:hex", "dep:futures-lite", "sqlx-d1-core/offline"]
+query = ["dep:sha2", "dep:hex", "dep:futures-lite", "sqlx-d1-core/offline"]
 
 [dependencies]
-sqlx-d1-core     = { path = "../sqlx-d1-core", version = "=0.3.1" }
-sqlx-core        = { version = "=0.8.3" }
-sqlx-macros-core = { version = "=0.8.3", features = ["macros"]}
-syn              = { version = "2.0", features = ["full"] }
-proc-macro2      = { version = "1.0" }
-quote            = { version = "1.0" }
-serde            = { version = "1.0", features = ["derive"] }
-serde_json       = { version = "1.0" }
-heck             = { optional = true, version = "0.5" }
-sha2             = { optional = true, version = "0.10" }
-hex              = { optional = true, version = "0.4" }
-futures-lite     = { optional = true, version = "2.6", features = ["std"], default-features = false }
+sqlx-d1-core = { path = "../sqlx-d1-core", version = "=0.4.0" }
+sqlx-core = { version = "=0.8.6" }
+sqlx-macros-core = { version = "=0.8.6", features = ["macros"] }
+syn = { version = "2.0", features = ["full"] }
+proc-macro2 = { version = "1.0" }
+quote = { version = "1.0" }
+serde = { version = "1.0", features = ["derive"] }
+serde_json = { version = "1.0" }
+heck = { optional = true, version = "0.5" }
+sha2 = { optional = true, version = "0.11" }
+hex = { optional = true, version = "0.4" }
+futures-lite = { optional = true, version = "2.6", features = [
+  "std",
+], default-features = false }

--- a/sqlx-d1/Cargo.toml
+++ b/sqlx-d1/Cargo.toml
@@ -16,8 +16,8 @@ license = { workspace = true }
 features = ["json", "uuid", "chrono", "decimal", "macros"]
 
 [dependencies]
-sqlx-d1-core   = { path = "../sqlx-d1-core",   version = "=0.3.1" }
-sqlx-d1-macros = { path = "../sqlx-d1-macros", version = "=0.3.1", optional = true }
+sqlx-d1-core = { path = "../sqlx-d1-core", version = "=0.4.0" }
+sqlx-d1-macros = { path = "../sqlx-d1-macros", version = "=0.4.0", optional = true }
 
 [features]
 json = ["sqlx-d1-core/json"]
@@ -41,7 +41,7 @@ DEBUG = ["sqlx-d1-core/DEBUG", "sqlx-d1-macros?/DEBUG"]
 ################################
 
 [dev-dependencies]
-worker = { version = "0.7", features = ["d1"] }
+worker = { version = "0.8", features = ["d1"] }
 serde = { version = "1.0", features = ["derive"] }
 
 [lints.clippy]

